application/src/main/webapp/js/chat-modules.js

```diff
@@ -389,11 +393,36 @@ ChatRoom.prototype.getUserLastMessage = function() {
 /**
  * Show Messages (json to html)
  * @param msgs : json messages data to show
+ * @param merge : if true merge msgs with the current list, else replace
  */
-ChatRoom.prototype.showMessages = function(msgs) {
+ChatRoom.prototype.showMessages = function(msgs, merge) {
```

Comment:
Do we really need this param "merge" ? I mean, we could check if this.messages is empty or not to know if we have to merge with existing messages, no ?